src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs

```diff
@@ -3050,171 +3025,159 @@ internal void Freeze()
             /// <summary>
             /// AVL rotate left operation.
             /// </summary>
-            /// <param name="tree">The tree.</param>
             /// <returns>The rotated tree.</returns>
-            private static Node RotateLeft(Node tree)
+            private Node RotateLeft()
             {
-                Requires.NotNull(tree, nameof(tree));
-                Debug.Assert(!tree.IsEmpty);
+                Debug.Assert(!this.IsEmpty);
+                Debug.Assert(!_right.IsEmpty);
                 Contract.Ensures(Contract.Result<Node>() != null);
 
-                if (tree._right.IsEmpty)
-                {
-                    return tree;
```

Comment:
This can never be hit because at the call sites we check that the balance factor of the parent tree is greater than zero, which means there must be at least 1 node on the right subtree. If you look at a code coverage report you'll see that this branch is never hit. So, I deleted it and replaced the if statement with an assert.